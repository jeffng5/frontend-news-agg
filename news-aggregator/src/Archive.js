import React, { useState } from 'react'
import './css/FrontPage.css'
import {Helpers} from "./helpers"
import myImage from './penfeather.png'


// let token = localStorage.getItem('res.token')
// const decode = jwtDecode(token)

// button component to archive article takes in data from parent component
const Archive = ({username, url, title,description, author}) =>{
    
    const [activeButton, setActiveButton] = useState(true)
    const [state, setState] = useState([])
  
    
    async function handleArchive() {
        apiCall();

    }

    //apiCall thru helper function to post article into archive table
    async function apiCall() {
        try {
        setActiveButton(false);
        const res = await Helpers.saveArticle(username,url,title, description,author)
        setState(res)
     

    }   catch (e) {
        console.log(e)
    }}

  
// if (decode)
    return (
        <>

        <button className='archive' onClick={handleArchive} style={{backgroundColor: activeButton ? 'gold' : 'grey' }}><img id= 'archive' src = {myImage} alt=''></img>Archive</button>


      
        </>
    )



}

export default Archive