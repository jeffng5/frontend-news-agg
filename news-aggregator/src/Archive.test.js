import React from 'react';
import { render } from '@testing-library/react';
import Archive from './Archive'

it('renders without crashing', function() {
  render(<Archive />);
})

//snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(<Archive />);
    expect(asFragment()).toMatchSnapshot();
  });


test('testing component', ()=> {
    const {getByText} = render( <Archive />);
    getByText('Archive')
})