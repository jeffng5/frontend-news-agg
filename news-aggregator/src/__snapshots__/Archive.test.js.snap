// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="archive"
    style="background-color: yellow;"
  >
    <img
      alt=""
      id="archive"
      src="penfeather.png"
    />
    Archive
  </button>
</DocumentFragment>
`;
